File: iot_card_platform/app/schemas/h5.py
"""
H5 自助服务相关模型
"""
from typing import Optional, List, Dict, Any, Union
from pydantic import BaseModel, Field


class H5PortalConfig(BaseModel):
    user_id: int
    user_name: str
    title: Optional[str] = None
    logo: Optional[str] = None
    banner: Optional[str] = None
    notice: Optional[str] = None
    contact_phone: Optional[str] = None
    contact_wechat: Optional[str] = None
    theme: Optional[Dict[str, Any]] = None
    allow_suspend: bool = True
    allow_resume: bool = True
    allow_remark: bool = True
    require_verify: bool = False
    status: str = "enabled"


class H5CardQueryRequest(BaseModel):
    keyword: str = Field(..., min_length=1, max_length=30, description="卡号/ICCID/后6位")


class H5CardCandidate(BaseModel):
    id: int
    iccid_masked: str
    msisdn_masked: Optional[str] = None
    status: Optional[str] = None
    status_name: Optional[str] = None
    spec_name: Optional[str] = None
    activated_at: Optional[str] = None
    expired_at: Optional[str] = None


class H5CardActionFlags(BaseModel):
    allow_suspend: bool = False
    allow_resume: bool = False
    allow_remark: bool = False


class H5CardDetail(BaseModel):
    card: Dict[str, Any]
    diagnostics: Optional[Dict[str, Any]] = None
    actions: H5CardActionFlags


class H5CardQueryResponse(BaseModel):
    match_type: str
    items: List[Union[Dict[str, Any], H5CardCandidate]]


class H5CardRemarkRequest(BaseModel):
    remark: str = Field(..., max_length=500)
    operator_name: Optional[str] = Field(None, max_length=50)
    operator_phone: Optional[str] = Field(None, max_length=20)


class H5CardSuspendRequest(BaseModel):
    reason: Optional[str] = Field(None, max_length=200)
    verify_code: Optional[str] = Field(None, max_length=20)


class H5CardResumeRequest(BaseModel):
    verify_code: Optional[str] = Field(None, max_length=20)

File: iot_card_platform/frontend/src/api/modules/h5.ts
/**
 * H5 自助服务 API
 */

import { get, post, put } from '@/utils/request'
import type { Card, CardDiagnostics } from '@/types/card'

export interface H5PortalConfig {
  user_id: number
  user_name: string
  title?: string
  logo?: string
  banner?: string
  notice?: string
  contact_phone?: string
  contact_wechat?: string
  theme?: Record<string, any>
  allow_suspend: boolean
  allow_resume: boolean
  allow_remark: boolean
  require_verify: boolean
  status: 'enabled' | 'disabled' | 'expired'
}

export interface H5CardCandidate {
  id: number
  iccid_masked: string
  msisdn_masked?: string
  status?: string
  status_name?: string
  spec_name?: string
  activated_at?: string
  expired_at?: string
}

export interface H5CardActions {
  allow_suspend: boolean
  allow_resume: boolean
  allow_remark: boolean
}

export interface H5CardDetail {
  card: Card
  diagnostics?: CardDiagnostics | null
  actions: H5CardActions
}

export interface H5CardQueryResponse {
  match_type: 'exact' | 'fuzzy_single' | 'fuzzy_multiple' | 'none'
  items: Array<H5CardDetail | H5CardCandidate>
}

export interface H5CardActionResult {
  card_id: number
  iccid: string
  action: 'suspend' | 'resume' | 'refresh' | 'device_separation'
  status: 'processing' | 'success' | 'failed' | 'unsupported'
  callback_no?: string
  suspend_callback_no?: string
  resume_callback_no?: string
  device_separation_detection_status?: 'detected' | 'clear' | 'pending' | 'unsupported'
  device_separation_detection_message?: string
  message?: string
}

export const h5Api = {
  getConfig(slug: string): Promise<H5PortalConfig> {
    return get<H5PortalConfig>(`/h5/${slug}/config`)
  },

  queryCard(slug: string, keyword: string): Promise<H5CardQueryResponse> {
    return post<H5CardQueryResponse>(`/h5/${slug}/card/query`, { keyword })
  },

  getCardDetail(slug: string, cardId: number): Promise<H5CardDetail> {
    return get<H5CardDetail>(`/h5/${slug}/card/${cardId}`)
  },

  suspendCard(slug: string, cardId: number, reason?: string): Promise<H5CardActionResult> {
    return post<H5CardActionResult>(`/h5/${slug}/card/${cardId}/suspend`, { reason })
  },

  resumeCard(slug: string, cardId: number): Promise<H5CardActionResult> {
    return post<H5CardActionResult>(`/h5/${slug}/card/${cardId}/resume`, {})
  },

  refreshCard(slug: string, cardId: number): Promise<H5CardActionResult> {
    return post<H5CardActionResult>(`/h5/${slug}/card/${cardId}/refresh`, {}, {
      timeout: 180000
    })
  },

  detectDeviceSeparation(slug: string, cardId: number): Promise<H5CardActionResult> {
    return post<H5CardActionResult>(`/h5/${slug}/card/${cardId}/device-separation`, {})
  },

  updateRemark(slug: string, cardId: number, remark: string, operatorName?: string, operatorPhone?: string): Promise<Card> {
    return put<Card>(`/h5/${slug}/card/${cardId}/remark`, {
      remark,
      operator_name: operatorName,
      operator_phone: operatorPhone
    })
  }
}
